packages/babel-parser/src/parser/expression.js

```diff
@@ -893,6 +901,16 @@ export default class ExpressionParser extends LValParser {
     return node;
   }
 
+  parseClassHasInstanceExpression(base, startPos, startLoc, state) {
+    this.next(); // eat `class`
+
+    const node = this.startNodeAt(startPos, startLoc);
+    node.instance = this.parseCallExpressionArguments(tt.parenR);
```

Comment:
The following cases are invalid
```js
class.hasInstance();
class.hasInstance(foo, bar);
class.hasInstance(...foo);
class.hasInstance(foo, ...bar);
```
we should throw accordingly.

Reply:
This problem has been fixed.

Reply:
> This problem has been fixed.

I don't think so. Can you add the examples in my previous comments as parser test cases? All of them should throw.

You can create a new folder in `packages/babel-parser/test/fixtures/experimental/class-brand-check`.